example_workspace/src/hello_world_example_app/app/build.gradle

```diff
@@ -10,8 +10,9 @@ android {
         externalNativeBuild {
             cmake {
                 arguments   '-DANDROID_STL=c++_shared',
-                            '-DCMAKE_INSTALL_PREFIX=' + System.getenv('CMAKE_PREFIX_PATH'),
-                            '-DCMAKE_FIND_ROOT_PATH=' + System.getenv('CMAKE_PREFIX_PATH'),
+                            '-DCMAKE_INSTALL_PREFIX=' + System.getenv('TARGET_DIR'),
+                            '-DCMAKE_FIND_ROOT_PATH=' + System.getenv('TARGET_DIR'),
+                            '-DCMAKE_PREFIX_PATH=' + System.getenv('TARGET_DIR'),
```

Comment:
Note to self: figure out if this extra variable is really necessary.

Reply:
I remember I set CMAKE_PREFIX_PATH, because if not CATKIN wasn't found (I will check that again).

Reply:
Please try with

```cmake
'-DCMAKE_FIND_ROOT_PATH=' + System.getenv('CMAKE_PREFIX_PATH'),
```

only, which is what worked for us in another project. Not sure whether Gradle would ever install packages, where `CMAKE_INSTALL_PREFIX` plays a role. But setting it also has an effect on `find_*()` commands because the `CMAKE_INSTALL_PREFIX` is implicitly added to `CMAKE_SYSTEM_PREFIX_PATH` and also searched for packages, libraries, headers, etc. after all paths in `CMAKE_PREFIX_PATH`.

All that is irrelevant when cross-compiling because `CMAKE_FIND_ROOT_PATH` is in effect: Relative to paths listed in `CMAKE_FIND_ROOT_PATH` all packages and libraries can be found at default locations without additionally setting `CMAKE_PREFIX_PATH`.